packages/core/src/fhirlexer/parse.ts

```diff
@@ -1,14 +1,17 @@
 import { TypedValue } from '../types';
 import { Token } from './tokenize';
-
+export interface AtomContext {
+  parent: AtomContext | undefined;
```

Comment:
Let's change this to `parent?: AtomContext` so it's optional, and `undefined` is implied.